client/src/context/ToastContext.tsx

```diff
@@ -0,0 +1,119 @@
+import type { PropsWithChildren } from 'react';
+import { createContext, useContext, useEffect, useState } from 'react';
+import styled, { css, keyframes } from 'styled-components';
+
+type ToastVariant = 'default' | 'success' | 'warning' | 'error';
+
+type ToastContextType = {
+  showToast: (variant: ToastVariant, message: string) => void;
+};
+
+const ToastContext = createContext<ToastContextType | undefined>(undefined);
+
+type ToastProviderPros = PropsWithChildren;
+
+export const ToastProvider = (props: ToastProviderPros) => {
+  const { children } = props;
+  const [toastState, setToastState] = useState<{ variant: ToastVariant; message: string | null }>({
+    variant: 'default',
+    message: null,
+  });
+
+  useEffect(() => {
+    if (toastState.variant && toastState.message !== null) {
+      const timer = setTimeout(() => {
+        setToastState({ variant: 'default', message: null });
+      }, 5000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [toastState]);
+
+  const showToast = (variant: ToastVariant, message: string) => {
+    setToastState({ variant, message });
+  };
```

Comment:
```suggestion
  const showToast = (message: string, options?: Partial<{ variant: ToastVariant }>) => {
    const { variant = 'default' } = options ?? {};
    setToastState({ variant, message });
  };
```

사용 케이스를 고려해보았을 때 `variant`는 일반적으로 `'default'`를 많이 사용하지 않을까요? 고니가 생각하기엔 어떤가요?

제 생각엔 굳이(?) 명시하지 않아도 바로 `'default'`로 사용할 수 있는 함수로 디자인한다면 더 괜찮을 것 같습니다!! suggestion을 참고해주세용

Reply:
```
'{ showToast: (message: string, options?: Partial<{ variant: ToastVariant; }> | undefined) => void; }' 형식은 'ToastContextType' 형식에 할당할 수 없습니다.
  'showToast' 속성의 형식이 호환되지 않습니다.
    '(message: string, options?: Partial<{    variant: ToastVariant;}>) => void' 형식은 '(variant: ToastVariant, message: string) => void' 형식에 할당할 수 없습니다.
      'options' 및 'message' 매개 변수의 형식이 호환되지 않습니다.
        'string' 유형에 'Partial<{ variant: ToastVariant; }>' 유형과 공통적인 속성이 없습니다.ts(2322)
index.d.ts(371, 9): 필요한 형식은 여기에서 'IntrinsicAttributes & ProviderProps<ToastContextType | undefined>' 형식에 선언된 'value' 속성에서 가져옵니다.

(property) React.ProviderProps<ToastContextType | undefined>.value: ToastContextType | undefined
```

위와 같은 오류가 나오네용...

Reply:
보류 하는 걸루...ㅎ.ㅎ